src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/MediationSuccessfulApplicantNotificationHandler.java

```diff
@@ -0,0 +1,79 @@
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.notify.NotificationService;
+import uk.gov.hmcts.reform.civil.notify.NotificationsProperties;
+import uk.gov.hmcts.reform.civil.prd.model.Organisation;
+import uk.gov.hmcts.reform.civil.service.OrganisationService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.utils.PartyUtils.getPartyNameBasedOnType;
+
+@Service
+@RequiredArgsConstructor
+public class MediationSuccessfulApplicantNotificationHandler extends CallbackHandler implements NotificationData {
+
+    private final NotificationService notificationService;
+    private final NotificationsProperties notificationsProperties;
+    private static final List<CaseEvent> EVENTS = List.of(CaseEvent.NOTIFY_APPLICANT_MEDIATION_SUCCESSFUL);
+    private static final String REFERENCE_TEMPLATE = "mediation-successful-applicant-notification-%s";
+    public static final String TASK_ID = "MediationSuccessfulNotifyApplicant";
+    private final OrganisationService organisationService;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT), this::notifyApplicant
+        );
+    }
+
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    private CallbackResponse notifyApplicant(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        notificationService.sendMail(
+            caseData.getApplicantSolicitor1UserDetails().getEmail(),
+            notificationsProperties.getNotifyApplicantLRMediationSuccessfulTemplate(),
+            addProperties(caseData),
+            String.format(REFERENCE_TEMPLATE, caseData.getLegacyCaseReference()));
+        return AboutToStartOrSubmitCallbackResponse.builder().build();
+
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public Map<String, String> addProperties(CaseData caseData) {
+
+        return Map.of(
+            CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference(),
+            CLAIM_LEGAL_ORG_NAME_SPEC, getApplicantLegalOrganizationName(caseData),
+            DEFENDANT_NAME, getPartyNameBasedOnType(caseData.getRespondent1())
+        );
+    }
+
+    private String getApplicantLegalOrganizationName(CaseData caseData) {
```

Comment:
We are repeating this method everywhere. Perhaps its better to put it in a service. this is what we have done with Felipe.

Reply:
Done, good point!